sktime/forecasting/arch.py

```diff
@@ -0,0 +1,405 @@
+    x : {np.ndarray, DataFrame}, optional
+        Exogenous regressors.  Ignored if model does not permit exogenous
+        regressors.
+
+    Hyper-parameters
+    ----------------
+    mean : str, optional
+        Name of the mean model.  Currently supported options are: 'Constant',
+        'Zero', 'LS', 'AR', 'ARX', 'HAR' and  'HARX'
+    lags : int or list (int), optional
+        Either a scalar integer value indicating lag length or a list of
+        integers specifying lag locations.
+    vol : str, optional
+        Name of the volatility model.  Currently supported options are:
+        'GARCH' (default), 'ARCH', 'EGARCH', 'FIARCH' and 'HARCH'
+    p : int, optional
+        Lag order of the symmetric innovation
+    o : int, optional
+        Lag order of the asymmetric innovation
+    q : int, optional
+        Lag order of lagged volatility or equivalent
+    power : float, optional
+        Power to use with GARCH and related models
+    dist : int, optional
+        Name of the error distribution.  Currently supported options are:
+            * Normal: 'normal', 'gaussian' (default)
+            * Students's t: 't', 'studentst'
+            * Skewed Student's t: 'skewstudent', 'skewt'
+            * Generalized Error Distribution: 'ged', 'generalized error"
+    hold_back : int
+        Number of observations at the start of the sample to exclude when
+        estimating model parameters.  Used when comparing models with different
+        lag lengths to estimate on the common sample.
+    rescale : bool
+        Flag indicating whether to automatically rescale data if the scale
+        of the data is likely to produce convergence issues when estimating
+        model parameters. If False, the model is estimated on the data without
+        transformation.  If True, than y is rescaled and the new scale is
+        reported in the estimation results.
+
+    See Also
+    --------
+    Autoregressive Integrated Moving Average (ARIMA) models
+
+    References
+    ----------
+    .. [1] Jason Brownlee. How to Model Volatility with ARCH and GARCH for Time
+       Series Forecasting.
+       https://machinelearningmastery.com/develop-arch-and-garch-models-for-time
+       -series-forecasting-in-python/
+    .. [2] GitHub repository of arch package (soft dependency).
+       https://github.com/bashtage/arch
+    .. [3] Documentation of arch package (soft dependency). Forecasting Volatility
+       with ARCH and it's variants.
+       https://arch.readthedocs.io/en/latest/univariate/introduction.html
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.arch import ARCH
+    >>> y = load_airline()
+    >>> forecaster = ARCH(
+    ...    mean="Constant",
+    ...    lags=0,
+    ...    vol="Garch",
+    ...    p=1,
+    ...    o=0,
+    ...    q=1,
+    ...    power=2.0,
+    ...    dist="Normal",
+    ...    hold_back=None,
+    ...    rescale=None,
+    ...):
```

Comment:
please make this shorter, maybe even default all params is possible

Reply:
Okay, I shall make all the parameters default.